core/src/main/java/org/apache/carbondata/core/carbon/querystatistics/DriverQueryStatisticsRecorder.java

```diff
@@ -78,106 +82,142 @@ public synchronized void recordStatisticsForDriver(QueryStatistic statistic, Str
    */
   public void logStatisticsAsTableDriver() {
     synchronized (lock) {
-      String tableInfo = collectDriverStatistics();
-      if (null != tableInfo) {
-        LOGGER.statistic(tableInfo);
+      for (String key: queryStatisticsMap.keySet()) {
+        // print sql_parse_t,load_meta_t,block_allocation_t,block_identification_t
+        // or just print block_allocation_t,block_identification_t
+        if (queryStatisticsMap.get(key).size() >= 2) {
```

Comment:
The call can return null, because the keyset obtained reflects the state of map when the .keySet() method was called, later changes in the map will not be reflected. So, if the element is removed in the meantime, get() can return null.

Solution: use an iterator over entrySet()